File: test/leetcode/string/vowel-spellchecker.test.ts
import { spellchecker } from '../../src/string/vowel-spellchecker';

describe('vowel spellchecker', () => {
  test('vowel spellchecker - test case 1', async () => {
    const wordlist = ['KiTe', 'kite', 'hare', 'Hare'];
    const queries = ['kite', 'Kite', 'KiTe', 'Hare', 'HARE', 'Hear', 'hear', 'keti', 'keet', 'keto'];

    expect(spellchecker(wordlist, queries)).toStrictEqual([
      'kite',
      'KiTe',
      'KiTe',
      'Hare',
      'hare',
      '',
      '',
      'KiTe',
      '',
      'KiTe'
    ]);
  });
});

File: test/leetcode/string/vowel_spellchecker_test.py
from leetcode.string.vowel_spellchecker import Solution


soln = Solution()


def test_case_1():
    wordlist = ["KiTe", "kite", "hare", "Hare"]
    queries = ["kite", "Kite", "KiTe", "Hare", "HARE", "Hear", "hear", "keti", "keet", "keto"]

    assert soln.spellchecker(wordlist, queries) == [
        "kite",
        "KiTe",
        "KiTe",
        "Hare",
        "hare",
        "",
        "",
        "KiTe",
        "",
        "KiTe",
    ]
